controllers/webspherelibertyapplication_controller.go

```diff
@@ -51,15 +51,15 @@ const applicationFinalizer = "finalizer.webspherelibertyapps.liberty.websphere.i
 
 // +kubebuilder:rbac:groups=security.openshift.io,resources=securitycontextconstraints,resourceNames=restricted,verbs=use,namespace=websphere-liberty-operator
 // +kubebuilder:rbac:groups=liberty.websphere.ibm.com,resources=webspherelibertyapplications;webspherelibertyapplications/status;webspherelibertyapplications/finalizers,verbs=get;list;watch;create;update;patch;delete,namespace=websphere-liberty-operator
-// +kubebuilder:rbac:groups=apps,resources=deployments;statefulsets,verbs=get;list;watch;create;update;patch;delete,namespace=websphere-liberty-operator
+// +kubebuilder:rbac:groups=apps,resources=deployments;statefulsets,verbs=list;watch;create;update;delete,namespace=websphere-liberty-operator
 // +kubebuilder:rbac:groups=apps,resources=deployments/finalizers;statefulsets,verbs=update,namespace=websphere-liberty-operator
-// +kubebuilder:rbac:groups=core,resources=services;secrets;serviceaccounts;configmaps;persistentvolumeclaims,verbs=get;list;watch;create;update;patch;delete,namespace=websphere-liberty-operator
-// +kubebuilder:rbac:groups=autoscaling,resources=horizontalpodautoscalers,verbs=get;list;watch;create;update;patch;delete,namespace=websphere-liberty-operator
-// +kubebuilder:rbac:groups=networking.k8s.io,resources=ingresses,verbs=get;list;watch;create;update;patch;delete,namespace=websphere-liberty-operator
-// +kubebuilder:rbac:groups=route.openshift.io,resources=routes;routes/custom-host,verbs=get;list;watch;create;update;patch;delete,namespace=websphere-liberty-operator
+// +kubebuilder:rbac:groups=core,resources=services;secrets;serviceaccounts;configmaps;persistentvolumeclaims,verbs=list;watch;create;update;delete,namespace=websphere-liberty-operator
```

Comment:
@yongja79 Thanks for the PR. 

`get` permission on Secret is needed for SSO (the relevant code is [here](https://github.com/WASdev/websphere-liberty-operator/blob/bc4cca50d4d01e72406e8ed56f412b6d000cf44d/utils/utils.go#L301).

So wondering why the SSO tests didn't fail even though the operator doesn't have permission to get Secret. 
Either these RBAC permissions are not used or the operator gets the permission via some other way. 

Deployment is the most used resource, so remove some/all obvious permissions for it to see if that results in any errors.